api/maven-api-core/src/main/java/org/apache/maven/api/Constants.java

```diff
@@ -0,0 +1,239 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api;
+
+import org.apache.maven.api.annotations.Config;
+
+/**
+ * Configuration constants.
+ */
+public final class Constants {
+
+    /**
+     * Maven home.
+     *
+     * @since 3.0.0
+     */
+    @Config(readOnly = true)
+    public static final String MAVEN_HOME = "maven.home";
+
+    /**
+     * Maven configuration.
+     *
+     * @since 3.0.0
+     */
+    @Config(defaultValue = "${maven.home}/conf")
+    public static final String MAVEN_CONF = "maven.conf";
+
+    /**
+     * Maven user home.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${user.home}/.m2")
+    public static final String MAVEN_USER_HOME = "maven.user.home";
+
+    /**
+     * Maven local repository.
+     *
+     * @since 3.0.0
+     */
+    @Config(defaultValue = "${maven.user.home}/repository")
+    public static final String MAVEN_REPO_LOCAL = "maven.repo.local";
+
+    /**
+     * Maven system-wide extensions.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${maven.conf}/extensions.xml")
+    public static final String MAVEN_SYSTEM_EXTENSIONS = "maven.system.extensions";
+
+    /**
+     * Maven user extensions.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${maven.user.home}/extensions.xml")
+    public static final String MAVEN_USER_EXTENSIONS = "maven.user.extensions";
+
+    /**
+     * Maven project-wide extensions.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${session.rootDirectory}/.mvn/extensions.xml")
+    public static final String MAVEN_PROJECT_EXTENSIONS = "maven.project.extensions";
+
+    /**
+     * Maven system toolchains.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${maven.conf}/toolchains.xml")
+    public static final String MAVEN_SYSTEM_TOOLCHAINS = "maven.system.toolchains";
+
+    /**
+     * Maven user toolchains.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "${maven.user.home}/toolchains.xml")
+    public static final String MAVEN_USER_TOOLCHAINS = "maven.user.toolchains";
+
+    /**
+     * Extensions class path.
+     */
+    @Config
+    public static final String MAVEN_EXT_CLASS_PATH = "maven.ext.class.path";
+
+    /**
+     * Maven output color mode.
+     * Allowed values are <code>auto</code>, <code>always</code>, <code>never</code>.
+     *
+     * @since 4.0.0
+     */
+    @Config(defaultValue = "auto")
+    public static final String MAVEN_STYLE_COLOR_PROPERTY = "maven.style.color";
+
+    /**
+     * Build timestamp format.
+     *
+     * @since 3.0.0
+     */
+    @Config(source = Config.Source.MODEL, defaultValue = "yyyy-MM-dd'T'HH:mm:ss'Z'")
+    public static final String MAVEN_BUILD_TIMESTAMP_FORMAT = "maven.build.timestamp.format";
```

Comment:
One of the problems with this is that the timezone (offset) is missing here. It should be either mentioned or we should switch to `yyyy-MM-dd'T'HH:mm:ssXXX` which will produce the same output as with `Etc/UTC`.